buildnDeploy/Dockerfile

```diff
@@ -0,0 +1,44 @@
+#This is a multistage docker file where build process is
+#seperated from the deployment
+
+# Base image of golang to build binaries of upspinserver
+FROM golang:latest as build
+
+WORKDIR /go/src
+
+# Clone the git repo of upspin
+RUN git clone https://github.com/upspin/upspin.git
+WORKDIR /go/src/upspin/upspin
+
+# Build the binaries
+RUN GOOS=linux GOARCH=amd64 go build upspin.io/cmd/upspinserver
```

Comment:
Currently the binaries are built in local disk, how do we address if the os is different so that the GOOS and GOARCH may change? An alternative condition is the storage is on Google Drive or cloud services.

Reply:
The binaries here will not be build in localdisk, they will be build in container itself, once they are build we are creating a new image by copying the binaries from previous stage.
The GOOS and GOARCH are parameters which indicates "on what system we want to run upspinserver", for us it will be always linux (containers are linux based) and amd64 (popular choice).